app/scripts/contentscript.js

```diff
@@ -106,14 +95,35 @@ function forwardTrafficBetweenMuxers(channelName, muxA, muxB) {
  * Error handler for page to extension stream disconnections
  *
  * @param {string} remoteLabel - Remote stream name
- * @param {Error} err - Stream connection error
+ * @param {Error} error - Stream connection error
  */
-function logStreamDisconnectWarning(remoteLabel, err) {
-  let warningMsg = `MetamaskContentscript - lost connection to ${remoteLabel}`
-  if (err) {
-    warningMsg += `\n${err.stack}`
-  }
-  console.warn(warningMsg)
+function logStreamDisconnectWarning(remoteLabel, error) {
+  console.debug(
+    `MetaMask Contentscript: Lost connection to "${remoteLabel}".`,
+    error,
+  )
+}
+
+/**
+ * This function must ONLY be called in pump destruction/close callbacks.
+ * Notifies the inpage context that streams have failed, via window.postMessage.
+ * Relies on obj-multiplex and post-message-stream implementation details.
+ */
+function notifyInpageOfStreamFailure() {
+  window.postMessage(
+    {
+      target: INPAGE, // the post-message-stream "target"
+      data: {
+        // this object gets passed to obj-multiplex
+        name: PROVIDER, // the obj-multiplex channel name
+        data: {
+          jsonrpc: '2.0',
+          method: 'METAMASK_STREAM_FAILURE',
+        },
+      },
+    },
+    window.location.origin,
+  )
```

Comment:
We use `postMessage` directly here because, by the time we want to send this message, the streams will have been torn down because they are all `pump`:ed together, and `pump` tears them all down if even a single stream fails.